packages/libs/sdk/src/solana/solana.ts

```diff
@@ -0,0 +1,140 @@
+// eslint-disable-next-line @nx/enforce-module-boundaries
+import {
+  Transaction,
+  Keypair,
+  ComputeBudgetProgram,
+  Connection,
+  TransactionInstruction,
+  PublicKey,
+  TransactionMessage,
+  VersionedTransaction,
+} from '@solana/web3.js';
+import {
+  type EstimatePriorityFeesParams,
+  type PriorityFeeRequestPayload,
+  type PriorityFeeResponseData,
+  type PriorityFeeLevels,
+} from './types';
+
+export class Solana {
+  readonly endpointUrl: string;
+  readonly connection: Connection;
+
+  constructor({ endpointUrl }: { endpointUrl: string }) {
+    this.endpointUrl = endpointUrl;
+    this.connection = new Connection(endpointUrl);
+  }
+
+  async sendWithPriorityFees(
+    transaction: Transaction,
+    keyPair: Keypair,
+    feeLevel: PriorityFeeLevels = 'medium'
+  ) {
+    // eslint-disable-next-line prefer-const
+    let [computeUnitPriceInstruction, units, recentBlockhash] =
+      await Promise.all([
+        this.createDynamicPriorityFeeInstruction(feeLevel),
+        this.getSimulationUnits(
+          this.connection,
+          transaction.instructions,
+          keyPair.publicKey
+        ),
+        this.connection.getLatestBlockhash(),
+      ]);
```

Comment:
the issue is that your simulation did not include the priority fee instruction…so it’s returning too few compute units. If you fetch the priority fee before the simulation and pass it into your simulation transaction it works:

```suggestion
  async sendWithPriorityFees(
    transaction: Transaction,
    keyPair: Keypair,
    feeLevel: PriorityFeeLevels = 'medium'
  ) {
    const computeUnitPriceInstruction = await this.createDynamicPriorityFeeInstruction(feeLevel);
    const allInstructions = [...transaction.instructions, computeUnitPriceInstruction];

    // eslint-disable-next-line prefer-const
    let [units, recentBlockhash] =
      await Promise.all([
        this.getSimulationUnits(
          this.connection,
          allInstructions,
          keyPair.publicKey
        ),
        this.connection.getLatestBlockhash(),
      ]);
```